extensions/android-paging3/src/main/java/com/squareup/sqldelight/android/paging3/QueryPagingSource.kt

```diff
@@ -59,11 +60,52 @@ internal abstract class QueryPagingSource<Key : Any, RowType : Any> :
 fun <RowType : Any> QueryPagingSource(
   countQuery: Query<Long>,
   transacter: Transacter,
-  dispatcher: CoroutineDispatcher,
+  dispatcher: CoroutineDispatcher = Dispatchers.IO,
   queryProvider: (limit: Long, offset: Long) -> Query<RowType>,
 ): PagingSource<Long, RowType> = OffsetQueryPagingSource(
   queryProvider,
   countQuery,
   transacter,
   dispatcher
 )
+
+/**
+ * Create a [PagingSource] that pages through results according to queries generated by
+ * [queryProvider]. Queries returned by [queryProvider] should expected to do keyset paging.
+ * For that reason, queries should be arranged by an non-ambigious `ORDER BY` clause. [Key] must
+ * be a unique clause that rows are ordered by.
+ *
+ * To support jumping [pageBoundaries] is a pre-calculated list of [Key] items that specifies
+ * where each page boundary exists within the full dataset. For example:
+ *
+ * The dataset `[0, 1, 2, 3, 4, 5, 6, 7, 8, 9]` ordered ascending with a page size of 2 would produce
+ * [pageBoundaries] `[0, 2, 4, 6, 8]`.
+ *
+ * The dataset `[0, 1, 2, 3, 4, 5, 6, 7, 8, 9]` ordered ascending with a page size of 3 would produce
+ * [pageBoundaries] `[0, 3, 6, 9]`.
+ *
+ * An example query returned by [queryProvider] could look like:
+ *
+ * ```sql
+ * SELECT value FROM numbers
+ * WHERE value >= 2
+ * ORDER BY value ASC
+ * LIMIT 2;
+ * ```
+ *
+ * Queries will be executed on [dispatcher].
+ */
```

Comment:
i was confused at the constructor, read this javadoc and it cleared everything up for me, well done

Reply:
Thanks for the review! this is still a WIP. It's close, though!

Reply:
I've uncommented the factory function for keyed paging and updated the docs. We shouldn't move the javadocs to the actual constructor of the class because it's internal. 

Let me know what you think of the documentation!

Reply:
whats the idea behind keeping the class internal?

Reply:
i see, because theres two types but just a single overloaded function. interesting

Reply:
For compatibility reasons. @JakeWharton wanted it hidden behind a factory function in the event that keyed paging was release _much later_ than offset paging. Hiding the implementations behind a function makes that easy.

Reply:
The class isn't useful except as an implementation of the interface so the function hides the class an exposes only the interface.